systemtest/src/test/java/io/strimzi/systemtest/mirrormaker/MirrorMaker2ST.java

```diff
@@ -1069,6 +1075,499 @@ void testRestoreOffsetsInConsumerGroup(ExtensionContext extensionContext) {
+            topicSourceNameA, kafkaClusterSourceName, messagesCount);
+        int sent = internalKafkaClient.sendMessagesTls();
+        internalKafkaClient.checkProducedAndConsumedMessages(sent, internalKafkaClient.receiveMessagesTls());
+
+        LOGGER.info("[DEFAULT] Changing to target - topic {}, cluster {}, user {} for next internalKafkaClient.",
+                topicTargetNameA, kafkaClusterTargetName, userTarget.getMetadata().getName());
+        internalKafkaClient = internalKafkaClient.toBuilder()
+            .withTopicName(topicTargetNameA)
+            .withClusterName(kafkaClusterTargetName)
+            .withKafkaUsername(userTarget.getMetadata().getName())
+            .withConsumerGroupName(ClientUtils.generateRandomConsumerGroup())
+            .build();
+
+        LOGGER.info("[DEFAULT] Now messages should be mirrored to target topic and cluster");
+        internalKafkaClient.checkProducedAndConsumedMessages(sent, internalKafkaClient.receiveMessagesTls());
+        LOGGER.info("[DEFAULT] Messages successfully mirrored");
+
+        Map<String, String> mmSnapshot = DeploymentUtils.depSnapshot(testStorage.getNamespaceName(), KafkaMirrorMaker2Resources.deploymentName(testStorage.getClusterName()));
+        LOGGER.info("[SCRAM-SHA] Changing KafkaUser sha-password on KMM2 Source and make sure it rolled");
+        Secret passwordSource = new SecretBuilder()
+            .withNewMetadata()
+            .withName(kafkaUserSourceName)
+            .endMetadata()
+            .addToData("password", "c291cmNlLXBhc3N3b3Jk")
+            .build();
+        kubeClient().patchSecret(testStorage.getNamespaceName(), kafkaUserSourceName, passwordSource);
+        mmSnapshot = DeploymentUtils.waitTillDepHasRolled(testStorage.getNamespaceName(), KafkaMirrorMaker2Resources.deploymentName(testStorage.getClusterName()), 1, mmSnapshot);
+
+        LOGGER.info("[SCRAM-SHA] Changing KafkaUser sha-password on KMM2 Target");
+        Secret passwordTarget = new SecretBuilder()
+            .withNewMetadata()
+            .withName(kafkaUserTargetName)
+            .endMetadata()
+            .addToData("password", "dGFyZ2V0LXBhc3N3b3Jk")
+            .build();
+        kubeClient().patchSecret(testStorage.getNamespaceName(), kafkaUserTargetName, passwordTarget);
+        mmSnapshot = DeploymentUtils.waitTillDepHasRolled(testStorage.getNamespaceName(), KafkaMirrorMaker2Resources.deploymentName(testStorage.getClusterName()), 1, mmSnapshot);
+
+        LOGGER.info("[SCRAM-SHA] Recreate kafkaClients pod with new passwords.");
+        resourceManager.deleteResource(kubeClient().namespace(testStorage.getNamespaceName()).getDeployment(testStorage.getKafkaClientsName()));
+        resourceManager.createResource(extensionContext, KafkaClientsTemplates.kafkaClients(testStorage.getNamespaceName(), true, testStorage.getKafkaClientsName(), userSource, userTarget).build());
```

Comment:
And also use it here (from previous comment)

Reply:
What about this?

Reply:
Not sure what you mean here. (feel free to PM me)